thorlcr/graph/thgraph.hpp

```diff
@@ -1175,7 +1192,9 @@ class graph_decl CActivityBase : implements CInterfaceOf<IThorRowInterfaces>, im
     CFileSizeTracker * queryTempFileSizeTracker()
     {
         if (!tempFileSizeTracker)
-            tempFileSizeTracker.setown(new CFileSizeTracker);
+        {
+            tempFileSizeTracker.setown(new CFileSizeTracker(queryGraph().queryTempFileSizeTracker()));
```

Comment:
I think this needs to use queryGraph().queryParent(), because in a child query, queryGraph() will be the child graph, not the subgraph that's tracking the usage. queryGraph().queryParent(), should always point to the subgraph, no matter what the depth of the nested child graphs.

NB: note to self - CGraphBase::queryParent() and CGraphBase::queryOwner() are counter-intuitive, it would have made more sense if queryOwner() was the top-level graph (the subgraph) and queryOwner() was the activity containing graph (whether that be the subgraph, or a nested child graph).

Do you have a spilling child graph test, you can test this with?

Reply:
I've attached loopsort2.ecl to the parent JIRA in case useful. It should spill in each iteration of the loop.
I've had a very brief look at the SizePeakTempDisk numbers:

```
sg1 (subgraph):
 <attr kind='SizePeakTempDisk' value='166813325' formatted='159.084MB' unit='sz' ctype='thor' creator='thor@172.26.29.236' ts='2024-07-17T08:40:07.780Z'/>

sg5 (child loop graph):
 <attr kind='SizePeakTempDisk' value='132142234' formatted='126.020MB' unit='sz' ctype='thor' creator='thor@172.26.29.236' ts='2024-07-17T08:40:07.780Z'/>

sg5:a7 (sort act in child graph)
 <attr kind='SizePeakTempDisk' value='29424207' formatted='28.060MB' unit='sz' ctype='thor' creator='thor@172.26.29.236' ts='2024-07-17T08:40:07.780Z'/>
```

I *think* the numbers should be broadly the same, i.e. it's not clear why sg5's number is significantly higher.
NB: the loop result may also spill, but I don't think it's currently being tracked.